### ISSUE TITLE
Adding third possible value for generate.subFolders other than true or false

### ISSUE THREAD

**Issue by fuweichin** [nuxt/nuxt#7053]
### What problem does this feature solve?

Problem:
No matter setting nuxt config generate.subFolders to true or false, the output directory layout is not so friendly with a web server like Nginx

Steps:
I have a project, which has the following structure
```txt
myapp/
  package.json
  nuxt.config.js
  pages/
    apps/
      index.vue
      _key.vue
```
In nuxt.config.js, if I set generate.subFolders to true, then generate static may look like this
```txt
myapp/
  dist/
    apps/
      index.html
      key1/
        index.html
      key2/
        index.html
```
or if to false, then something else
```txt
myapp/
  dist/
    apps.html
    apps/
      key1.html
      key2.html
```
I cannot say it's the problem, but It may be better if a third solution which which can be more friendly with existing web servers, like Nginx,HTTPD.

### What does the proposed changes look like?

Possible Solution:

1. Adding third possible value "unset" for generate.subFolders
2. With generate.subFolders set to "unset", then output directory layout may looks like 
```txt
myapp/
  dist/
    apps/
      index.html
      key1.html
      key2.html
```
In short, transform without applying special rules.

apps/index.vue => apps/index.html
apps/_key.vue => apps/key1.html, apps/key2.html

Thus developers can use Nginx directive try_files like this to serve a Nuxt static site

```nginx
  try_files $uri.html $uri/ $uri =404;
```

Such then it will work perfect.

**Comment by danielroe** [comment]
In Nuxt 3 you can customise the output path with the `prerender:generate` Nitro hook, which I think is a reasonable approach for this. If you think we still need this, do let me know and we can reconsider.
